frontend/packages/knative-plugin/src/utils/create-eventsources-utils.ts

```diff
@@ -264,59 +254,26 @@ export const sanitizeKafkaSourceResource = (formData: EventSourceFormData): Even
   };
 };
 
-export const getEventSourceConnectorList = (
+export const getEventSourceList = (
   namespace: string,
-  csvData: ClusterServiceVersionKind[],
-) =>
-  _.reduce(
-    csvData,
-    (acm, cv) => {
-      const parseCSVData = parseALMExamples(cv, false);
-      _.forEach(parseCSVData, (res) => {
-        if (
-          _.findIndex(acm, res) === -1 &&
-          referenceFor(res) === referenceForModel(EventSourceCamelModel) &&
-          res?.metadata?.labels?.[EVENT_SOURCE_LABEL] === 'true'
-        ) {
-          const { apiGroup, plural, kind } = modelFor(referenceFor(res));
-          const {
-            metadata: { name },
-          } = res;
-          const modelData = {
-            [name]: {
-              name,
-              iconUrl: getEventSourceIcon(kind, res),
-              displayName: _.startCase(name),
-              title: _.startCase(name),
-              data: { almData: res },
-            },
-          };
-          acm.push(
-            checkAccess({
-              group: apiGroup,
-              resource: plural,
-              namespace,
-              verb: 'create',
-            }).then((result) => (result.status.allowed ? modelData : {})),
-          );
-        }
-      });
-      return acm;
-    },
-    [],
-  );
-
-export const getEventSourceList = (namespace: string, eventSourceModels: K8sKind[]) => {
+  eventSourceModels: K8sKind[],
+  t: TFunction,
+) => {
   const accessList = [];
   eventSourceModels.map((model) => {
-    const { apiGroup, plural, kind } = model;
+    const { apiGroup, plural, kind, label: name, id: uid } = model;
+    const href = `/catalog/ns/${namespace}/eventsource?sourceKind=${kind}`;
+    const { description, provider } = getEventSourceCatalogProviderData(kind, t) ?? {};
     const modelData = {
-      [kind]: {
-        name: kind,
-        iconUrl: getEventSourceIcon(kind),
-        displayName: _.startCase(kind),
-        title: _.startCase(kind),
-        provider: isKnownEventSource(kind) ? 'Red Hat' : '',
+      uid,
+      name,
+      description,
+      icon: { url: getEventSourceIcon(referenceForModel(model)), class: null },
+      type: 'EventSource',
+      provider,
+      cta: { label: t('knative-plugin~Create Event Source'), href },
+      data: {
+        sourceType: kind,
```

Comment:
It would be better to keep this logic separate from this util. if we keep it here this would affect the catalog and quick search as well. I think we should just keep the normalization logic separate in the provider itself and return models from here like we did before